core/domain/suggestion_registry.py

```diff
@@ -162,6 +163,21 @@ def validate(self):
                     'Expected final_reviewer_id to be a string, received %s' %
                     type(self.final_reviewer_id))
 
+        if (
+                not user_id_migration.verify_user_id_correct(self.author_id) and
```

Comment:
I don't think we should add this validation in the domain layer, I would rather suggest doing this in the service layer before storing the suggestion and in the controller layer while fetching request data. [Also, make sure to have this as a prod data validation.]

Reply:
I added a domain validation check in the services layer before the PUT. The domain validation will also be included in the prod data validation checks.